src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -405,36 +405,45 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
         return new Tuple<bool, string>(success, paymentIntentClientSecret);
     }
 
-    private async Task ValidateSecretsManagerSeatsAndServiceAccount(OrganizationUpgrade upgrade, Organization organization, short newPlanSeats, Models.StaticStore.Plan newSecretsManagerPlan)
+    private async Task ValidateSecretsManagerSeatsAndServiceAccountAsync(OrganizationUpgrade upgrade, Organization organization,
+        Models.StaticStore.Plan newSecretsManagerPlan)
     {
         var newPlanSmSeats = (short)(newSecretsManagerPlan.BaseSeats +
                                      (newSecretsManagerPlan.HasAdditionalSeatsOption
                                          ? upgrade.AdditionalSmSeats
                                          : 0));
+        var occupiedSmSeats =
+            await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
+
         if (!organization.SmSeats.HasValue || organization.SmSeats.Value > newPlanSmSeats)
         {
-            var occupiedSmSeats =
-                await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
-            if (occupiedSmSeats > newPlanSeats)
+            if (occupiedSmSeats > newPlanSmSeats)
             {
                 throw new BadRequestException(
                     $"Your organization currently has {occupiedSmSeats} Secrets Manager seats filled. " +
-                    $"Your new plan only has ({newPlanSeats}) seats. Remove some users.");
+                    $"Your new plan only has ({newPlanSmSeats}) seats. Remove some users.");
             }
         }
 
+        var occupiedPmSeats = await _organizationUserRepository.GetOccupiedSeatCountByOrganizationIdAsync(organization.Id);
+
+        if (occupiedSmSeats > occupiedPmSeats)
+        {
+            throw new BadRequestException("Your secrets manager seats should not be greater than password manager occupied seat");
+        }
+
         if (newSecretsManagerPlan.BaseServiceAccount != null)
         {
             var newPlanServiceAccount = (short)(newSecretsManagerPlan.BaseServiceAccount +
                                                 (newSecretsManagerPlan.HasAdditionalServiceAccountOption
-                                                    ? upgrade.AdditionalServiceAccount
+                                                    ? upgrade.AdditionalServiceAccounts
                                                     : 0));
             if (!organization.SmServiceAccounts.HasValue ||
                 organization.SmServiceAccounts.Value > newPlanServiceAccount)
             {
                 var occupiedServiceAccount =
                     await _serviceAccountRepository.GetServiceAccountCountByOrganizationIdAsync(organization.Id);
-                if (occupiedServiceAccount > newPlanSeats)
+                if (occupiedServiceAccount > newPlanSmSeats)
```

Comment:
Don't use `occupied` to refer to service accounts. You can call it `currentServiceAccounts`. 

Also, you need to compare to `plan.MaxServiceAccount`, not `newPlanSmSeats`.

Reply:
Okay, But I will check with Max the value for the MaxService account for all the Plan Types because is actually null for all the secrets manager plan type

Reply:
It only applies to the free SM plan, which has a max of 3 service accounts. That's already in the `SecretsManagerPlanStore`. I think it's correct that it's null for all other plans, but always safer to double check if you want.